vpr/src/route/router_lookahead_map_utils.cpp

```diff
@@ -175,18 +208,18 @@ void expand_dijkstra_neighbours(util::PQ_Entry parent_entry,
             continue;
         }
 
-        util::PQ_Entry child_entry(child_node_ind, switch_ind, parent_entry.delay,
-                                   parent_entry.R_upstream, parent_entry.congestion_upstream, false);
+        util::PQ_Entry_Lite child_entry(child_node_ind, switch_ind, parent_entry.delay_cost, false);
 
-        VTR_ASSERT(child_entry.cost >= 0);
+        VTR_ASSERT(child_entry.delay_cost >= 0);
 
-        /* skip this child if it has been visited with smaller cost */
-        if (node_visited_costs[child_node_ind] >= 0 && node_visited_costs[child_node_ind] < child_entry.cost) {
+        /* skip this child if it has been visited with smaller or the same cost */
```

Comment:
You are querying the map twice, once on line 216 and once on line 222.  You can avoid this double query by changing the find on 216 to an insert.  If the entry doesn't, it will be inserted, which would've been done anyways on line 222.  Then the result of insert can be queried to determine if the insert failed.  If the insert failed, you can check if the child_entry cost is lower.

Reply:
Fixed.